### PR TITLE
script to transform loket to lmb

### PR DESCRIPTION
**Pull request by Rahien**
## Description

This script transforms a loket db to a database that can be used to create a toLoad for an LMB database. The how to is in the readme

## How to test

Start from a dump from a loket db. Run it in your virtuoso (don't run our migrations). Then run this script. Once done, create a dump file and transform it into a toLoad folder. Use that for starting a new instance of the LMB db.

### CODE REVIEW

File: scripts/loket-to-lmb-transform/deleteUnwantedInstances.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+export async function deleteUnwantedInstances(){
+  console.log("Deleting unwanted instances...");
+  await deleteUnwantedBestuursOrgs();
+  await deleteUnwantedMandates();
+  await deleteUnwantedMandatarissen();
+  await deleteUnwantedPersons();
+  await deleteUnwantedBestuurseenheden();
+  await deleteUnwantedGeboorten();
+  await deleteUnwantedIdentifiers();
```
Comment (bfidlers) [inline scripts/loket-to-lmb-transform/deleteUnwantedInstances.ts:192]: Not sure if we delete everything that is unwanted, what about memberships en rechtstreekseVerkiezingen that are linked to resources that are deleted here (I think the specific resouces we delete won't have these relations, but I'm not entirely sure)

**Review by bfidlers — APPROVED** [review]
Few remarks here and there, but nothing mayor that would result in a noticably different result I think.

File: scripts/loket-to-lmb-transform/dropUnimportantGraphs.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  console.log(`Dropped all unimportant graphs!`);
+}
+
+async function getUnimportantGraphs(){
+  const result = await querySudo(`
+    SELECT DISTINCT ?g WHERE {
+      GRAPH ?g {
+        ?s ?p ?o
+      }
+      FILTER (?g != <http://mu.semte.ch/graphs/public> && !STRENDS(STR(?g), "/LoketLB-mandaatGebruiker") && !STRENDS(STR(?g), "/LoketLB-LeidinggevendenGebruiker"))
```
[the remark this answers is not in the document]
Comment (Rahien) [reply scripts/loket-to-lmb-transform/dropUnimportantGraphs.ts:21]: correct! removing!

File: scripts/loket-to-lmb-transform/moveTypesOutOfPublic.ts [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+  } WHERE {
+    GRAPH <http://mu.semte.ch/graphs/public> {
+      ?bestuurseenheid a besluit:Bestuurseenheid;
+                      mu:uuid ?bestuurseenheidUuid;
+                      ^besluit:bestuurt ?orgaan.
+      ?orgaan a besluit:Bestuursorgaan.
+      {
+        ?orgaan ?p ?o.
+        VALUES ?p { rdf:type mu:uuid skos:prefLabel lmb:deactivatedAt
+                    besluit:bestuurt ext:origineleBestuurseenheid besluit:classificatie ext:origineleBestuursorgaan }
```
[the remark this answers is not in the document]
Comment (Rahien) [reply scripts/loket-to-lmb-transform/moveTypesOutOfPublic.ts:41]: fair. but it doesn't hurt.

File: scripts/loket-to-lmb-transform/moveTypesOutOfPublic.ts [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+    FILTER(?g != <http://mu.semte.ch/graphs/public>)
+
+    ?orgaan ^mandaat:isTijdspecialisatieVan ?orgaanInTijd.
+
+    GRAPH <http://mu.semte.ch/graphs/public> {
+      ?orgaanInTijd a besluit:Bestuursorgaan.
+      {
+        ?orgaanInTijd ?p ?o.
+        VALUES ?p { rdf:type mu:uuid dct:modified lmb:deactivatedAt mandaat:bindingEinde mandaat:bindingStart
+                    mandaat:isTijdspecialisatieVan org:hasPost lblodlg:heeftBestuursfunctie ext:heeftBestuursperiode }
```
[the remark this answers is not in the document]
Comment (Rahien) [reply scripts/loket-to-lmb-transform/moveTypesOutOfPublic.ts:78]: I'll add dct modified everywhere and remove the heeftBestuursfunctie :+1:

File: scripts/loket-to-lmb-transform/moveTypesOutOfPublic.ts [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+}
+
+export async function moveOutOfPublic() {
+  console.log("moving types out of public:");
+  await moveBestuursorgaan();
+  await moveBestuursorgaanInTijd();
+  await moveMandaat();
+  await moveVerkiezing();
+  await moveKandidatenLijst();
+  await moveVerkiezingsResultaat();
```
[the remark this answers is not in the document]
Comment (Rahien) [reply scripts/loket-to-lmb-transform/moveTypesOutOfPublic.ts:314]: by just using inserts it seems virtuoso can handle it without batching. It seems to get confused if insert/delete queries touch triples that it removed during the query.

File: scripts/loket-to-lmb-transform/deleteUnwantedInstances.ts (hunk 1 above)
Comment (Rahien) [reply scripts/loket-to-lmb-transform/deleteUnwantedInstances.ts:192]: Memberships are for fractions which only exist for our domain (not for erediensten etc that these queries target). Same for verkiezingen

**Pull request merged**